### DISCUSSION TITLE
No column named translations

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#15261] [Archive]
### Describe the Bug

Having a table names "meals" with a bunch of fields and added a "translations" field from directus.
When using the frontend I see the browser is sending this payload:
```
{
   "id":"123",
   "translations":{
	"create":[
		{
			"name":"Hi",
			"languages_code":{
				"code":"de-DE"
			}
		}
	],
         "update":[],
         "delete":[]
     }
}
```

which seems totaly fine for me. But writing a hook in the backend where i call `database(tablename).insert(payload)` I recieve the following error:

```
directusrocketmeals | [Error: insert into `meals` (`id`, `status`, `translations`) values ('6', 'published', '{"create":[{"name":"Kartoffelsalat","languages_code":{"code":"de-DE"}},{"name":"Potato salad","languages_code":{"code":"en-US"}}]}') - SQLITE_ERROR: table meals has no column named translations] {
directusrocketmeals |   errno: 1,
directusrocketmeals |   code: 'SQLITE_ERROR'
directusrocketmeals | }
```

### To Reproduce

I can share my repo if this helps.

### Errors Shown

```
directusrocketmeals | [Error: insert into `meals` (`id`, `status`, `translations`) values ('6', 'published', '{"create":[{"name":"Kartoffelsalat","languages_code":{"code":"de-DE"}},{"name":"Potato salad","languages_code":{"code":"en-US"}}]}') - SQLITE_ERROR: table meals has no column named translations] {
directusrocketmeals |   errno: 1,
directusrocketmeals |   code: 'SQLITE_ERROR'
directusrocketmeals | }
```

### What version of Directus are you using?

9.12.2

### What version of Node.js are you using?

v18.0.0

### What database are you using?

sqlite

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by NilsBaumgartner1994** [comment]
Ah okay it creates a M2M relation ... thought it would be easier, but since i only have a knex instance i need to manualy link it.

**Comment by azrikahar** [comment]
The browser payload may not actually be representative of a payload that can be plugged into a knex insert statement, so we can't exactly do such operation directly. In this case, particularly the`create/update/delete` syntax:

```
{
	"create":[
		{
			"name":"Hi",
			"languages_code":{
				"code":"de-DE"
			}
		}
	],
         "update":[],
         "delete":[]
}
```

that is primarily for ItemsService (and the PayloadService within ItemsService) to process, and knex wouldn't really understand it 👍

**Comment by NilsBaumgartner1994** [comment]
Maybe using ItemService: https://github.com/directus/directus/discussions/9859
